File: F/3159-find-occurrences-of-an-element-in-an-array/main.go
func occurrencesOfElement(nums []int, queries []int, x int) []int {
	pos := []int{}
	for i, v := range nums {
		if v == x {
			pos = append(pos, i)
		}
	}

	ans := []int{}
	for _, q := range queries {
		if q > len(pos) {
			ans = append(ans, -1)
		} else {
			ans = append(ans, pos[q-1])
		}
	}
	return ans
}

File: F/3159-find-occurrences-of-an-element-in-an-array/main.rs
struct Solution {}

impl Solution {
  pub fn occurrences_of_element(nums: Vec<i32>, queries: Vec<i32>, x: i32) -> Vec<i32> {
    let filter = nums
      .into_iter()
      .enumerate()
      .filter(|v| v.1 == x)
      .collect::<Vec<(usize, i32)>>();
    queries.iter().map(|&q| {
      if q <= filter.len() as i32  {
        filter[q as usize-1].0 as i32
      } else {
        -1
      }
    }).collect::<Vec<i32>>()
  }
}

fn main() {
  println!(
    "{:?}",
    Solution::occurrences_of_element(vec![1, 3, 1, 7], vec![1, 3, 2, 4], 1)
  );
}
